### PR TITLE
#47 re design bike upload

### LINKED ISSUE #47: ReDesign Bike upload to incorporate support for image analysis from back-end
Basically add a upload button that kicks the image stored to the server. Wait for a reply before submit can be done. The server should return a url to the uploaded image which must be included.

### PR DESCRIPTION
**Pull request by FredSoderberg**
#### What does this PR do?
Redesigned our bike upload to thunks and to match our backend. Sadly backend must adapt though since keyword hierarchy needed is not possible without modifying backend.

#47 (LINK_TO_ISSUE)

##### Why are we doing this? Any context or related work?
Connect upload bike to the rest of the app

#### Where should a reviewer start?
addbike.js and check the submit option, follow redux actions there and then into addbikeactions

#### Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).
closes #47

### CODE REVIEW

File: Bikeini/navigation/actions/AddBikeActions.js [hunk 1]
```diff
@@ -7,15 +17,86 @@ export const saveImageToState = uri => (
...
+    type: 'image/jpg',
+    // TODO: FIX NAME!!!
+    name: `${bikeData.type}.jpg`,
+  };
+  const formBody = new FormData();
+  formBody.append('image', file);
+  // Object.entries(bikeData).map(([key, value]) => formBody.append(key, value));
+  // Object.entries(bikeData.location).map(([key, value]) => formBody.append('location', value));
+  // Object.entries(bikeData.keywords).map(([key, value]) => formBody.append('keywords', value));
+  formBody.append('json', JSON.stringify(bikeData));
```
Comment (jakobemilandersson) [inline Bikeini/navigation/actions/AddBikeActions.js:95]: Should the comments still be here? 🐱

File: Bikeini/screens/Login.js [hunk 2]
```diff
@@ -70,6 +70,7 @@ class Login extends React.Component {
     const {
       authState, profileState, navigation, loadProfileInit,
     } = this.props;
+    // console.log(profileState, authState);
```
Comment (jakobemilandersson) [inline Bikeini/screens/Login.js]: console.log 🌵 (comment tho)

File: Bikeini/screens/Login.js [hunk 3]
```diff
@@ -78,7 +79,9 @@ class Login extends React.Component {
       && !profileState.loadingProfile
       && !profileState.error
       && !authState.isLoggedIn
-      && !authState.deletingJwt) {
+      && !authState.deletingJwt
+      && !authState.error) {
+      // console.log('LOADING PROFILE');
```
Comment (jakobemilandersson) [inline Bikeini/screens/Login.js]: commented console 💯 idc about comments tho 🥇

**Review by jakobemilandersson — APPROVED** [review]

**Comment by jakobemilandersson** [comment]
Everything looks good, have a nice weekend 🕺

File: Bikeini/navigation/actions/AddBikeActions.js (hunk 1 above)
Comment (FredSoderberg) [reply Bikeini/navigation/actions/AddBikeActions.js:95]: nahh but fixed

**Pull request merged**